File: OpenEdXMobile/src/main/java/org/edx/mobile/authentication/LoginTask.java
package org.edx.mobile.authentication;

import android.content.Context;
import androidx.annotation.NonNull;

import com.google.inject.Inject;

import org.edx.mobile.task.Task;

public abstract class LoginTask extends Task<AuthResponse> {

    @NonNull
    private final String username;
    @NonNull
    private final String password;

    @Inject
    private LoginAPI loginAPI;

    public LoginTask(@NonNull Context context, @NonNull String username, @NonNull String password) {
        super(context);
        this.username = username;
        this.password = password;
    }

    @Override
    @NonNull
    public AuthResponse call() throws Exception {
        return loginAPI.logInUsingEmail(username, password);
    }
}

File: OpenEdXMobile/src/main/java/org/edx/mobile/authentication/LoginTask.kt
package org.edx.mobile.authentication

import android.content.Context
import dagger.hilt.android.EntryPointAccessors
import org.edx.mobile.core.EdxDefaultModule.ProviderEntryPoint
import org.edx.mobile.model.authentication.AuthResponse
import org.edx.mobile.task.Task

abstract class LoginTask(
    context: Context,
    private val username: String,
    private val password: String
) : Task<AuthResponse?>(context) {

    var loginAPI = EntryPointAccessors
        .fromApplication(context, ProviderEntryPoint::class.java).getLoginAPI()

    override fun doInBackground(vararg voids: Void): AuthResponse? {
        try {
            return loginAPI.logInUsingEmail(username, password)
        } catch (e: Exception) {
            e.printStackTrace()
            handleException(e)
        }
        return null
    }
}
